### ISSUE TITLE
linter/no-unused-vars: should take into account JSDoc `{@link MyType}`

### ISSUE THREAD

**Issue by mesqueeb** [oxc-project/oxc#12824]
with code:
```ts
import type { MyType } from './main'

/** also check out {@link MyType} */
const t = 1
```
we get
```
Type 'MyType' is imported but never used.
help: Consider removing this import.oxc[eslint(no-unused-vars)](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-unused-vars.html)
```
but it should know that `MyType` is being used in the JSDoc link (you can click on `MyType` when hovering over `t` in this example.)

**Comment by Sysix** [comment]
another issue with the same problem: #11639

**Comment by taearls** [comment]
It doesn't appear that eslint supports marking jsdoc variable references as used, either. It's not in their [documentation](https://eslint.org/docs/latest/rules/no-unused-vars) for this rule

**Comment by camc314** [comment]
I have been thinking about this a decent amount the last week, and I think i know how i want to solve this, unfortunately, it's non trivial

**Comment by Sysix** [comment]
another issue: #12945

**Comment by camc314** [comment]
Closing as a duplicate of #11639. This is the same `no-unused-vars` false positive for an imported type that is only referenced from a JSDoc/TSDoc `{@link ...}` tag.


Thanks!
